examples/hawkbit-device-simulator/src/main/java/org/eclipse/hawkbit/simulator/DeviceSimulatorUpdater.java

```diff
@@ -85,38 +121,131 @@ public void startUpdate(final String tenant, final String id, final long actionI
         private final long actionId;
         private final EventBus eventbus;
         private final UpdaterCallback callback;
+        private final List<SoftwareModule> modules;
 
         private DeviceSimulatorUpdateThread(final AbstractSimulatedDevice device, final SpSenderService spSenderService,
-                final long actionId, final EventBus eventbus, final UpdaterCallback callback) {
+                final long actionId, final EventBus eventbus, final UpdaterCallback callback,
+                final List<SoftwareModule> modules) {
             this.device = device;
             this.spSenderService = spSenderService;
             this.actionId = actionId;
             this.eventbus = eventbus;
             this.callback = callback;
+            this.modules = modules;
         }
 
         @Override
         public void run() {
+            if (device.getProgress() <= 0 && modules != null) {
+                device.setUpdateStatus(simulateDownloads(device.getTargetSecurityToken()));
+                if (device.getUpdateStatus().getResponseStatus().equals(ResponseStatus.ERROR)) {
```

Comment:
If you use `if(ResponseStatus.ERROR.equals(device.getUpdateStatus().getResponseStatus())` then you have a implict null check for the getResponseStatus()